src/streamlink/plugins/huya.py

```diff
@@ -72,52 +79,79 @@ def _get_streams(self):
                             {
                                 "sCdnType": str,
-                                "iPCPriorityRate": int,
                                 "sStreamName": str,
                                 "sFlvUrl": str,
                                 "sFlvUrlSuffix": str,
                                 "sFlvAntiCode": validate.all(str, validate.transform(html_unescape)),
                             },
                             validate.union_get(
                                 "sCdnType",
-                                "iPCPriorityRate",
                                 "sStreamName",
                                 "sFlvUrl",
                                 "sFlvUrlSuffix",
                                 "sFlvAntiCode",
                             )),
                         ],
                     }],
+                    "vMultiStreamInfo": [
+                        {
+                            'iBitRate': int
+                        }
+                    ]
                 },
-                validate.get(("data", 0)),
                 validate.union_get(
-                    ("gameLiveInfo", "liveId"),
-                    ("gameLiveInfo", "nick"),
-                    ("gameLiveInfo", "roomName"),
-                    "gameStreamInfoList",
+                    ('data', 0, "gameLiveInfo", "liveId"),
+                    ('data', 0, "gameLiveInfo", "nick"),
+                    ('data', 0, "gameLiveInfo", "roomName"),
+                    ('data', 0, "gameStreamInfoList"),
+                    'vMultiStreamInfo'
                 ),
             ),
         ))
         if not data:
             return
 
-        self.id, self.author, self.title, streamdata = data
+        self.id, self.author, self.title, streamdata, vMultiStreamInfo = data
 
         self.session.http.headers.update({
             "Origin": "https://www.huya.com",
             "Referer": "https://www.huya.com/",
         })
 
-        for cdntype, priority, streamname, flvurl, suffix, anticode in streamdata:
-            qs = {k: v for k, v in dict(parse_qsl(anticode)).items() if k in self._STREAM_URL_QUERYSTRING_PARAMS}
-            url = update_scheme("https://", f"{flvurl}/{streamname}.{suffix}")
-            url = update_qsd(url, qs)
-
-            name = f"source_{cdntype.lower()}"
-            self.QUALITY_WEIGHTS[name] = self._QUALITY_WEIGHTS_OVERRIDE.get(name, priority)
-            yield name, HTTPStream(self.session, url)
+        for cdntype, streamname, flvurl, suffix, anticode in streamdata:
+            for vStreamInfo in vMultiStreamInfo:
+                iBitRate = vStreamInfo['iBitRate']
+                qs = {k: v for k, v in dict(parse_qsl(anticode)).items() if k in self._STREAM_URL_QUERYSTRING_PARAMS}
+                url = update_scheme("http://", f"{flvurl}/{streamname}.{suffix}")
+                params = self._get_stream_params(qs['fm'], qs['fs'], qs['ctype'], qs['wsTime'], streamname, iBitRate)
```

Comment:
The `qs` / `params` solution isn't ideal because of the `__getitem__()` lookups from the "anticode" dict, which can raise `KeyError`s. This was not the case previously, because the whole (filtered) dict was simply serialized as a query string to the URL.

We usually avoid any potential exceptions by using validation schemas, which print much better error messages, but this is a bit unnecessary here. Just using `.get("...", default_value)` should be good enough in order to avoid `KeyError`s.